pkg/database/interface.go

```diff
@@ -0,0 +1,115 @@
+package database
+
+import (
+	"context"
+	"crypto/sha256"
+	"time"
+
+	"github.com/codenotary/immudb/embedded/sql"
+	"github.com/codenotary/immudb/embedded/store"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	schemav2 "github.com/codenotary/immudb/pkg/api/schemav2"
+)
+
+type DB interface {
```

Comment:
good to have interface definition on its own file

Reply:
moved